### ISSUE TITLE
absolute polarization calibration & polselfcal discussion

### ISSUE THREAD

**Issue by o-smirnov** [ratt-ru/CubiCal#329]
Starting this as a catch-all for this topic, but also so I can show Rick some plots.

**Comment by o-smirnov** [comment]
## Ku-band relative leakages as computed by Rick & AIPS

|D|C|BnA|A1|A2|
|-|-|-|-|-|
|![U-D-MID-POLSOLN](https://user-images.githubusercontent.com/6470079/68480820-62f9fe00-023e-11ea-9e04-90efaa088d08.png)|![U-C-MID-POLSOLN](https://user-images.githubusercontent.com/6470079/68480823-63929480-023e-11ea-90ea-3a230943c47e.png)|![U-BNA-MID-POLSOLN](https://user-images.githubusercontent.com/6470079/68480824-63929480-023e-11ea-9b03-6d6413e215d0.png)|![U-A1-MID-POLSOLN](https://user-images.githubusercontent.com/6470079/68480826-642b2b00-023e-11ea-9fca-0e8e0e69283f.png)|![U-A2-MID-POLSOLN](https://user-images.githubusercontent.com/6470079/68480825-63929480-023e-11ea-8b0f-b9bcf65aa243.png)|

## Differences w.r.t. A1 leakages

|A1-D|A1-C|A1-BnA|A1-A2|
|-|-|-|-|
|![U-D-MID-POLSOLN diff](https://user-images.githubusercontent.com/6470079/68480960-ace2e400-023e-11ea-870b-b9c12455da58.png)|![U-C-MID-POLSOLN diff](https://user-images.githubusercontent.com/6470079/68480962-ace2e400-023e-11ea-8523-c3f7b44c9508.png)|![U-BNA-MID-POLSOLN diff](https://user-images.githubusercontent.com/6470079/68480963-ad7b7a80-023e-11ea-8ada-c7cf603be26b.png)|![U-A2-MID-POLSOLN diff](https://user-images.githubusercontent.com/6470079/68480964-ad7b7a80-023e-11ea-86bd-6da0ac95500c.png)|

**Comment by o-smirnov** [comment]
And this is our whiteboard from the discussion. Very glad we had that meeting, as it made me finally understand why absolute leakages are so hard to get at, and why convergence is so finicky:
 
![image](https://user-images.githubusercontent.com/6470079/68481809-d997fb00-0240-11ea-9b56-3a8352dabac5.png)

**Comment by o-smirnov** [comment]
So @JSKenyon, thinking it through from the PoV of the solver. Imagine we have a source of 1 Jy, with linear pol ~0.1, and the d terms are also ~0.1.

In the cross-hand visibilities (e.g. at element 12), you then have a sum of the following components. Think of a vector sum on the complex plane:

1. The true polarization component, P, which rotates with P.A. This is ~0.1.

2. First-order leakage from I, given by I.(dpr + dql*). This doesn't rotate. This is also ~0.1.

3. Second-order leakage from the other polarization, P*.dpr.dql*, which counter-rotates with PA. This is ~0.001, so let's forget about it.

The traditional approach is to ignore the parallel hands, and to just use (2) to solve for leakages. ~0.1 is plenty of signal. But because dpr+dql* always appear in pairs, there's an ambiguity -- one could add any X to all the d-rights, and -X* to all the d-lefts, and get the same answer. We therefore pick a reference antenna, set its d-right to 0, and get all the other leakages relative to that. 

To get an absolute D solution, we need to look at the parallel hands. This is a sum of:

1. Stokes I, which doesn't rotate, and is 1

2. I.dpr.dql*, which doesn't rotate, and is ~0.01 in amplitude (NB: and is the reason we really want to know the true i.e. absolute not relative Ds in the first place)

3. P.dqr* and P*.dpr, which rotate and counter-rotate with PA, and are ~0.01 in amplitude.

So the "driving term" for the absolute Ds is only 0.01, an order of magnitude lower than the one in the crosshands. Furthermore, the data is not perfectly calibrated. A 1deg error in phase and a 1% error in amplitude will already give you an offset in the parallel-hand data that is already comparable to the driving term. But if you solve for G before you solve for D, you start absorbing leakages into G. Vice versa is also problematic, especially if any antennas have a systematic amplitude error -- this will immediately bias their D solutions.

Hence, my naive approach of simply solving for a D-Jones only works well with data that is 

(a) very well-calibrated using a nearby reference calibrator (Rick's old 3C286+3C287 observation)

(b) long and plentiful, so that the residual calibration errors in I average out over the entire observation (which is the solution interval for the Ds).   

This doesn't seem to be the case for the newer 3C286+OQ observations, so I've got convergence issues.

**Comment by o-smirnov** [comment]
Actually, it's not that bad -- in the 3C286+OQ case, after a lot of fiddling with Jones chains and ``term-iters``, I can get decent-looking leakage solutions for most antennas (and by decent, I mean they match those from AIPS, once the reference antenna is subtracted out). But on a few antennas, these leakages are higher than AIPS by some scaling factor (1.5~2) that seems constant across frequency. This makes me suspect that there's a corresponding scaling factor being pulled into the G solutions (which have a frequency interval of 0).

Things to be checked then (making a list for my own benefit):

0. Look at G solutions for the offending antennas to see if the above is the case.

1. Add a reference antenna option to the leakage solver. This is a one-liner. Combining that with ``-offdiag-only 1``, it should be able to solve for relative Ds using the cross-hands only, in the traditional manner.

2. Verify that the solutions from (1) match those that Rick gets out of AIPS.

Then, a germ of a further idea. If we establish reliable relative Ds using (1), we're left with a single unknown offset. That's one complex number per solution interval, across all antennas. Why not have a specialized machine that solves for this offset, @JSKenyon? I.e. work out a Jones parameterization of the form

Dp = [[1, dpr + x],[dql - x*,1]]

...where dpr, dql are loaded from the previous step, and x is just the one common unknown complex offset. I think its Wirtinger Jacobian should be straightforward to work out. 

I think this can potentially work better than the current approach, because the driving terms would be accumulated across all times *and* all antennas, thus averaging out the residual calibration errors a lot better.

**Comment by JSKenyon** [comment]
@o-smirnov I have started to take a look at the maths for this but I am not sure that I understand the parameterisation you suggested above. In my mind, if we are looking for a phase offset, it will be multiplicative not additive (as suggested by the whiteboard photo). Additionally, the parameterisation for Dp should not include a component from Dq. Thus, my instinct is that our parameterisation is:

Dp = [ [1, dpr*x], [dpl*x.conj, 1] ] and Dq = [ [1, dqr*x], [dql*x.conj, 1] ]

where our goal is to solve for the x (and technically x.conj). Let me know if I have completely lost it.

**Comment by JSKenyon** [comment]
As per our discussion, I did have my wires crossed. Reminder to myself that the x we are looking for is to move us from a relative leakage solution to an absolute leakage solution.

**Comment by bennahugo** [comment]
I'm not convinced either of the above are correct. In both linear and circular basis I think you need 
Dp = [[x, x.dpr],[dyp, 1]] 
x is **phase only** to line up the relative phases across the entire array. See https://arxiv.org/pdf/1706.06612.pdf

**Comment by bennahugo** [comment]
where x and y is interchangeable for r and l

**Comment by bennahugo** [comment]
dxi and dyi is complex although the phases are residuals and noise like from my experience with MeerKAT.

**Comment by o-smirnov** [comment]
> I'm not convinced either of the above are correct. 

Yeah I was obviating the crosshand phase term there (thinking of it as a separate matrix), which needs to be solved for separately. But I already have code there to estimate it.

If we roll all terms into D, the full parameterization is

D<sub>p</sub> = [ [e<sup>iX</sup>, d<sub>pr</sub>+A], [d<sub>pl</sub>-A&#773;, 1] ] 

Where X is the common cross-hand phase offset (aka the PZD), and A is the common additive leakage offset (have renamed it from small x to avoid ambiguity with big X, which is already staked out for cross-hand phase in the literature).

X can be further parameterized as a frequency slope.

The idea is to first solve for the X and the complex d's using the off-diagonal terms, then to load those solutions, and solve for A using the full matrix.

And of course rl can be xy.
